front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -11,16 +17,146 @@ import { useMemo } from "react";
 
 import DataSourceResourceSelectorTree from "@app/components/DataSourceResourceSelectorTree";
 import { useParentResourcesById } from "@app/hooks/useParentResourcesById";
+import { orderDatasourceViewByImportance } from "@app/lib/assistant";
 import {
   CONNECTOR_CONFIGURATIONS,
   getConnectorProviderLogoWithFallback,
 } from "@app/lib/connector_providers";
-import { getDisplayNameForDataSource, isFolder } from "@app/lib/data_sources";
+import {
+  getDisplayNameForDataSource,
+  isFolder,
+  isManaged,
+  isWebsite,
+} from "@app/lib/data_sources";
+
+const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
+const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+
+type DataSourceViewsSelectorProps = {
+  owner: WorkspaceType;
+  dataSourceViews: DataSourceViewType[];
+  selectionConfigurations: DataSourceViewSelectionConfigurations;
+  setSelectionConfigurations: Dispatch<
+    SetStateAction<DataSourceViewSelectionConfigurations>
+  >;
+};
+
+export function DataSourceViewsSelector({
+  owner,
+  dataSourceViews,
+  selectionConfigurations,
+  setSelectionConfigurations,
+}: DataSourceViewsSelectorProps) {
+  const applyGrouping =
```

Comment:
This code could benefits from some comments to explain what we are trying to do.